src/WebJobs.Extensions.DurableTask/EndToEndTraceHelper.cs

```diff
@@ -51,11 +66,12 @@ public static string LocalSlotName
             string version,
             string instanceId,
             string reason,
-            bool isOrchestrator,
+            FunctionType functionType,
             bool isReplay)
         {
-            EtwEventSource.Instance.FunctionScheduled(hubName, LocalAppName, LocalSlotName, functionName, version, instanceId, reason, isOrchestrator, isReplay);
-            this.appTraceWriter.Info($"[DF] {instanceId}: Scheduling function '{functionName}', version '{version}'. reason: {reason}. IsReplay: {isReplay}.");
+            EtwEventSource.Instance.FunctionScheduled(hubName, LocalAppName, LocalSlotName, functionName, version, instanceId, reason, functionType.ToString(), isReplay);
+            this.logger.LogInformation("{instanceId}: Function '{functionName} ({functionType})', version '{version}' scheduled. Reason: {reason}. IsReplay: {isReplay}. State: {state}. HubName: {hubName}. AppName: {appName}. SlotName: {slotName}.", 
+                instanceId, functionName, functionType, version, reason, isReplay, FunctionState.Scheduled, hubName, LocalAppName, LocalSlotName);;
```

Comment:
Should these be using `LogDebug()`? These seem like higher-level debug logs that people may not want if they've set up 'Information' as their default.

Reply:
These are analytic logs that will be important to have enabled always. Ideally they would be in a separate table so that they don't get mixed with application logs, but I assume that isn't possible?

Reply:
Thinking about this more, I do worry about the potential App Insights costs associated with this analytic tracing. @tohling do you think we should add a config switch to allow customers to *disable* this if they think it's too expensive? @christopheranderson I'd love to get your thoughts on this as well.

Reply:
@cgillum, yes a config switch to opt-in enabling AppInsights logging is a good approach.

Reply:
Switches are fine, as long as it is turned on by default. Don't stress about pricing. We'll also eventually support Azure Monitoring logs which is cheaper, so design for what customers need.

Reply:
Yes, please make it _opt-out_ (i.e. enabled by default). It's a real drag when folks need this data to diagnose issues and we have to teach them how to turn it on. I think it's better that we teach them how to turn it off after they've carefully considered their diagnostic needs. I'm glad to know we'll support cheaper options in the future.

Reply:
Confirmed with ChrisG about this and we are in agreement that currently, we want this to be LogInformation.  We may change this in the future if that is problematic.

---
In reply to: [136400005](https://github.com/Azure/azure-functions-durable-extension/pull/37#discussion_r136400005) [](ancestors = 136400005)